### PR TITLE
Radiation: ccnorm and cloud random overlap options

### PR DESCRIPTION
**Pull request by brianhenn**
Allows the radiation port to operate with cloud random overlap namelist option in addition to pre-existing maximum-random overlap option. Also allows the port to scale condensate by cloud fraction (`ccnorm=True`). As always, the radiation port's options are specified to be the same as the Fortran radiation scheme's options in prognostic runs. Also improves some of the docstrings the radiation wrapper. 

Added public API:
- radiation port can be run with GFS physics namelist options `ccnorm`: `True` and `iovr_sw`/`iovr_lw`: `0`.

- [X] Tests added

Coverage reports (updated automatically):
- test_unit: [82%](https://output.circle-artifacts.com/output/job/84e459c2-e8fb-4e16-9df2-764a959f737c/artifacts/0/tmp/coverage/htmlcov-test_unit/index.html)

### CODE REVIEW

File: external/radiation/radiation/config.py
Comment (frodre) [inline external/radiation/radiation/config.py:176]: Was there a specific need for moving the arg declaration into the dict which is then read back in via the arg generation function? I think it might be adding complexity without much benefit.  Perhaps these are used somewhere outside this module?

I like the previous method of initiating the class by listing dataclass args with the corresponding value from the namelist.  That way if any refactors happen it's immediately apparent that the init arguments need to change as well instead of redirecting around this file to find changes need to happen in the module-level dict.  I would also consider adding a `.from_namelist` to `GFSPhysicsControl` so that we can just call that in the `RadiationConfig.from_namelist`.  This would similarly confine any updates of `GFSPhysicsControl` to just that class.

**Review by frodre — CHANGES_REQUESTED** [review]
Thanks, @brianhenn!  In general, looks good.  Thanks for the documentation updates and tests included.  

I think the `config.py` changes make things more complicated than necessary, but happy to jump on a call to discuss or let me know if I'm missing something.  After that point is resolved, happy to approve.

**Review by frodre — APPROVED** [review]
Approved after discussion w/ @brianhenn on `config.py` changes

File: external/radiation/tests/test_config.py
Comment (brianhenn) [reply external/radiation/tests/test_config.py:2]: This is a good idea but I'm holding off for now because the radiation driver and wrapper are not set up in a way that allows for easy mocking/testing. Some refactors to disentangle the I/O from the code we'd want to test will be coming in a future PR I think.

**Pull request merged**
